tests/test_knn_classifier.py

```diff
@@ -115,7 +115,7 @@ def test_predict(self) -> None:
             len([self.sample_input]),
             self.model.training_inputs.shape[0],
         )
-        assert labels.shape == (top_k, len([self.sample_input]))
+        assert labels.shape == (len([self.sample_input]),)
         assert similar_samples.shape == (top_k, len([self.sample_input]))
```

Comment:
This may be a separate issue, and it's not new with this PR, but `len([self.sample_input])` is always 1, because it is taking the length of the 1-element list `[self.sample_input]`. So the assertions here are equivalent to:

- `assert distance.shape == (1, self.model.training_inputs.shape[0])`
- `assert labels.shape == (1,)` – previously `assert labels.shape == (top_k, 1)`
- `assert similar_samples.shape == (top_k, 1)`

It's not obvious to me whether that was intended. @zbloss

Reply:
No you're right, updating to be more clear on what exactly is being tested here.